### PR TITLE
Tdl 17 split03

### PR DESCRIPTION
**Pull request by trifrog**
Split previous merge request tdl_17

### CODE REVIEW

File: test/34.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+const expect = require('chai').expect;
+//let solution = require('../solutions/34').solution;
+solution = require('../yourSolution').solution;
+
+describe('repeating elements', () => {
+  it.only('1 should return repeating elements of an array', () => {
```
Comment (msach22) [inline test/34.js]: remove .only
that should only be there for you to test locally

File: test/34.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    const result = solution([1,2,2,3],3);
+    expect(result).to.deep.equal(1);
+  });
+
+  it.only('2 should return repeating elements of an array', () => {
+    const result = solution([1,2,2,3],2);
```
Comment (msach22) [inline test/34.js]: remove .only

File: test/34.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    expect(result).to.deep.equal(2);
+  });
+
+  it.only('3 should return repeating elements of an array', () => {
+    const result = solution([1,2,2,3],9);
```
Comment (msach22) [inline test/34.js]: remove .only

File: test/10.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+const expect = require('chai').expect;
+let solution = require('../solutions/10').solution;
+solution = require('../yourSolution').solution;
+
+describe('Test 1: Given a non-prime number, test should fail.', () => {
+  it.only('Result should return false', () => {
```
Comment (msach22) [inline test/10.js]: remove .only

File: test/10.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    const result = solution(8);
+    expect(result).to.deep.equal(false);
+  });
+});
+
+
+describe('Test 2: Given a prime number, test should pass.', () => {
+  it.only('Result should return true', () => {
```
Comment (msach22) [inline test/10.js]: remove .only

File: solutions/10.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
+// Name
+// Problem Description
+
+const solution = (numberToCheck) => {
+
```
Comment (songz) [inline solutions/10.js]: delete empty lines

File: solutions/10.js [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+  for(i=2; i<numberToCheck; i++){
```
Comment (songz) [inline solutions/10.js]: for(let i=2; ...

File: test/10.js [hunk 8]
```diff
@@ -0,0 +1,18 @@
+const expect = require('chai').expect;
+let solution = require('../solutions/10').solution;
+solution = require('../yourSolution').solution;
```
Comment (songz) [inline test/10.js]: comment out

File: solutions/10.js [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+        return false;
+      }
+    }
+    return true;
+  }
+}
+
+const solution = (arr) => {
+  let sumOfPrimeNumbers = 0;
+  for(let i = 0; i < arr.length; i++){
```
Comment (songz) [inline solutions/10.js]: in ES6, the best way to write a for loop is:

```
arr.forEach((e) => {
  ...
});
```

File: solutions/10.js [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+    if(isPrime(arr[i])){
+      sumOfPrimeNumbers = sumOfPrimeNumbers + arr[i];
+    } 
+  }
+  return sumOfPrimeNumbers;
+};
+
+console.log("f", solution([1,2,3,9,8,7]));
```
Comment (songz) [inline solutions/10.js]: `remove console.log`

File: test/10.js [hunk 11]
```diff
@@ -0,0 +1,10 @@
+const expect = require('chai').expect;
+let solution = require('../solutions/10').solution;
+//solution = require('../yourSolution').solution;
+
+describe('Test 1: Given [1,2,3,9,8,7]', () => {
```
Comment (songz) [inline test/10.js]: Add more test case. Particularly edge cases that might break your app. Always test for edge cases.

I would add these: `[1], [3], [1,3]`

File: solutions/10.js [hunk 12]
```diff
@@ -0,0 +1,34 @@
+// Tri Le
+// Find sum of prime numbers 
+
+const isPrime = (number) => {
+  if(number === 2){
```
Comment (songz) [inline solutions/10.js]: 2 is already supported in your for loop, So this if statement would be duplicate code

File: solutions/10.js [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+    return true;
+  }
+  else if (number < 2){
+    return false;
+  }
+  else {
```
Comment (songz) [inline solutions/10.js]: Don't need else statement here. Return stops the function, so if the code is still running, none of the cases matched.

File: solutions/10.js [hunk 14]
```diff
@@ -0,0 +1,35 @@
...
+    if(number%i === 0){
+      return false;
+    }
+  }
+  return true;
+}
+
+const solution = (arr) => {
+  let sumOfPrimeNumbers = 0;
+  console.log("identifying value of arr is:", arr);
```
Comment (songz) [inline solutions/10.js]: console.log.

File: solutions/10.js [hunk 15]
```diff
@@ -0,0 +1,34 @@
+// Tri Le
+// Find sum of prime numbers 
+
+const isNumber = (element) => {
```
Comment (songz) [inline solutions/10.js]: `typeof(a) === 'number'`

File: test/10.js [hunk 16]
```diff
@@ -0,0 +1,22 @@
...
+  });
+  it('should return sum as 0 for [1].', () => {
+    const result = solution([1]);
+    expect(result).to.equal(0);
+  });
+  it('should return sum as 2 for [-1.0.1.2.4.6.8.9].', () => {
+    const result = solution([-1,0,1,2,4,6,8,9]);
+    expect(result).to.equal(2);
+  });
+  it("should return summ as 9 for ['A','B','C',2,7].", () => {
```
Comment (songz) [inline test/10.js]: Not exactly relevant to the question.
